test/iaux-underlined-tab-bar.test.ts

```diff
@@ -0,0 +1,95 @@
+import { html, fixture, expect, oneEvent } from '@open-wc/testing';
+
+import type { UnderlinedTabBar } from '../src/iaux-underlined-tab-bar';
+import '../src/iaux-underlined-tab-bar';
+
+describe('UnderlinedTabBar', () => {
+  it('has a list tag', async () => {
+    const el = await fixture<UnderlinedTabBar>(
+      html`<iaux-underlined-tab-bar></iaux-underlined-tab-bar>`
+    );
+
+    const list = el.shadowRoot?.querySelector('ul');
+    expect(list).to.exist;
+  });
+
+  it('has a list tag', async () => {
+    const el = await fixture<UnderlinedTabBar>(
+      html`<iaux-underlined-tab-bar
+        .entries=${[{ displayName: 'UPLOADS' }, { displayName: 'POSTS' }]}
+      ></iaux-underlined-tab-bar>`
+    );
+
+    const list = el.shadowRoot?.querySelectorAll('li');
+    expect(list?.length).to.equal(2);
+  });
+
+  it('has no passes', async () => {
+    const el = await fixture<UnderlinedTabBar>(
+      html`<iaux-underlined-tab-bar></iaux-underlined-tab-bar>`
+    );
+
+    const list = el.shadowRoot?.querySelector('li');
+    expect(list).to.not.exist;
+  });
+
+  it('emits an event when item is clicked on', async () => {
+    const el = await fixture<UnderlinedTabBar>(
+      html`<iaux-underlined-tab-bar
+        .entries=${[{ displayName: 'UPLOADS' }, { displayName: 'POSTS' }]}
+      ></iaux-underlined-tab-bar>`
+    );
+
+    const button = el.shadowRoot?.querySelector('button');
+    const listener = oneEvent(el, 'itemclicked');
+    button?.click();
+    const { detail } = await listener;
+    expect(detail.index).to.equal(0);
+  });
+
+  it('adds active class to item clicked', async () => {
+    const el = await fixture<UnderlinedTabBar>(
+      html`<iaux-underlined-tab-bar
+        .entries=${[{ displayName: 'UPLOADS' }, { displayName: 'POSTS' }]}
+      ></iaux-underlined-tab-bar>`
+    );
+
+    const button = el.shadowRoot?.querySelector('button');
+    button?.click();
+    await el.updateComplete;
+    expect(button?.classList.contains('active')).to.be.true;
```

Comment:
when I am testing that something changes, I like to first assert the initial state (eg that initially the class list does _not_ contain `active`) which helps show that the action triggered the desired change (ie that the button click changed added the `active` class)

Reply:
The button is set to active by default, so I added a test to make sure the class was set before the click interaction